### PR TITLE
Helper methods for deleting file groups with their contents

### PR DESCRIPTION
**Pull request by evgkirov**
## Description

Related issue: #198 

> Motivated by https://github.com/uploadcare/pyuploadcare/issues/196
> 
> Groups are unremovable. But the files in the group are. Some times user has a need to remove all files in a group and currently, the only way to do this is to iterate over files and use delete or batch delete method.
> This makes code more complex and, potentially, increases number of requests to achieve the goal.
> 
> p.s.: Also, consider adding this as a core REST API functionality?

Changing the Uploadcare API is out of scope of this project. Helpers methods, however, have been implemented.

Example usage for `Uploadcare`:

```
uploadcare.delete_file_group_with_files("2e481e82-2e39-41be-a5a2-23802c6e341c~2") 
```

Example usage for `FileGroup`:

```
group = uploadcare.file_group("2e481e82-2e39-41be-a5a2-23802c6e341c~2")
group.delete_with_files()
```

## Checklist

- [x] Tests (if applicable)
- [x] Documentation (if applicable)
- [x] Changelog stub (or use [conventional commit messages](https://www.conventionalcommits.org/))

### CODE REVIEW

File: pyuploadcare/resources/file_group.py [hunk 1]
```diff
@@ -223,3 +223,12 @@ def delete(self):
 
         self._client.groups_api.delete(self.id)
         self._is_deleted = True
+
+    def delete_with_files(self):
```
[the remark this answers is not in the document]
Comment (evgkirov) [reply pyuploadcare/resources/file_group.py:227]: Can be either way.

[the remark this answers is not in the document]
Comment (dmitry-mukhin) [reply pyuploadcare/resources/file_group.py:227]: let's do this, "delete_with_files" is ugly.
Same for client method.

[the remark this answers is not in the document]
Comment (evgkirov) [reply pyuploadcare/resources/file_group.py:227]: I agree. 

And this functionality should be stripped down from the client as it doesn't have the regular `delete_file_group` method.

The argument name will be `delete_files` if that's ok:
```
group = uploadcare.file_group("2e481e82-2e39-41be-a5a2-23802c6e341c~2")
group.delete(delete_files=True)
```

[the remark this answers is not in the document]
Comment (dmitry-mukhin) [reply pyuploadcare/resources/file_group.py:227]: ok!

[the remark this answers is not in the document]
Comment (evgkirov) [reply pyuploadcare/resources/file_group.py]: Done

**Review by dmitry-mukhin — APPROVED** [review]

**Pull request merged**